ThunderCloud/gunzip.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  gunzip.swift
+//  ThunderCloud
+//
+//  Created by Simon Mitchell on 07/04/2020.
+//  Copyright © 2020 threesidedcube. All rights reserved.
+//
+
+import Foundation
+import zlib
+
+/// Gunzips a file, without reading the whole file to memory
+/// - Parameters:
+///   - fileURL: The file url of the file to gunzip
+///   - destination: The destination to gunzip the file to
+public func gunzip(_ fileURL: URL, to destination: URL) throws {
+    
+    guard let writeStream = OutputStream(toFileAtPath: destination.path, append: false) else {
+        throw GunzipError.failedToCreateWriteStream
+    }
+    
+    let cfWriteStream = writeStream as CFWriteStream
+    // Open the write stream
+    guard CFWriteStreamOpen(writeStream) else {
+        throw GunzipError.failedToOpenWriteStream
+    }
```

Comment:
Do we want to `defer { CFWriteStreamClose(cfWriteStream) }` here in case there is an error before the below `defer`?

Reply:
Yes, I may not do it as a defer though, as I'm not sure in which order two `defer` statements will be called!